### ISSUE TITLE
PSD2 Challenge view back pressed

### ISSUE THREAD

**Issue by DJuanan** [Adyen/adyen-android#212]
Every time the user press "back" in the PSD2 challenge view, the caller activity receives the next sequence:
**onResume --> onPause --> onResume**
That is causing me some problems with the error handling.

This is not happening when the user presses the "cancel" top-right button in the challenge view.

**Comment by caiofaustino** [comment]
By PDS2 you mean the 3DS2 SDK? Are you using the 3DS2 Component?

**Comment by DJuanan** [comment]
Exactly 3DS2. and yes, 3DS2 Component

**Comment by caiofaustino** [comment]
Are you using the error observer to check for errors and cancelation?
`adyen3DS2Component.observeErrors(lifecycleOwner, observer)`
You will get back a `ComponentError` and you can check if `componentError.getException()`  it's an instance of `Cancelled3DS2Exception`

**Comment by DJuanan** [comment]
Yes, I'm doing it already and I get the `Cancelled3DS2Exception`, but the behavior when you return to the previous activity is not the same if you press cancel or you do a "back", causing in the previous activity the strange sequence of calls I wrote up there.

**Comment by caiofaustino** [comment]
Hmmm, that's strange, but might require an update on the 3DS2 SDK instead. I'll have to investigate that better.
Do you get the calls in this order?
onResume --> onError --> onPause --> onResume
or 
onResume --> onPause --> onResume --> onError
